crates/sui-core/src/authority.rs

```diff
@@ -1433,25 +1432,102 @@ impl AuthorityState {
             .map(|handler| handler.event_store.clone())
     }
 
-    /// Returns a set of events corresponding to a given transaction, in order events were emitted
-    pub async fn get_events_for_transaction(
+    /// Returns events emitted in the given transaction, in order of events were emitted
+    /// Number of returned items are capped to EVENT_STORE_QUERY_MAX_LIMIT
+    pub async fn get_events_by_transaction(
         &self,
         digest: TransactionDigest,
-    ) -> Result<Vec<StoredEvent>, SuiError> {
+        limit: usize,
+    ) -> Result<Vec<SuiEventEnvelope>, anyhow::Error> {
         let es = self.get_event_store().ok_or(SuiError::NoEventStore)?;
-        es.events_for_transaction(digest).await
+        let stored_events = es.events_by_transaction(digest, limit).await?;
+        StoredEvent::into_event_envelopes(stored_events)
     }
 
-    /// Returns a whole set of events for a range of time
-    pub async fn get_events_for_timerange(
```

Comment:
can you give one example of how this api can be used for debugging? I can't think of one so decided to remove it.

Reply:
It's really simple.  We want to see all the events for the last day, or the last hour, or some given hour.  This might just be to check and see what kinds of events are being generated, to look at more than one account, etc.   

It's roughly equivalent to the dump tool from @oxade ..  I think this will be really useful for debugging Event Store.